dali/operators/audio/nonsilence_op.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/static_switch.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(NonsilenceRegion)
+                .DocStr(R"code(The operator performs leading and trailing silence detection in an audio buffer.
+This operators' behaviour can be described as::
+
+  def nonsilence(buffer, cutoff_value, top_db, ref_db):
+      buffer = ToDecibels(buffer, top_db, ref_db)
+      begin = end = 0
+      for i in [0 .. len(buffer)):
+          if buffer[i] > cutoff_value:
+              begin = i; break
+      for i in [len(buffer)-1 .. begin]:
+          if buffer[i] > cutoff_value:
+              end = i; break
+      length = end - begin + 1
+      return begin, length
+
+
+
++----------------------+-----------------------------------------+
```

Comment:
This table could be a overkill in the doc.

Reply:
I admit, creating this table isn't the easiest thing in `rst`, but I think it does great work with describing what are the inputs and outputs to the operator, what we don't actually have right now. I'd keep it, it really looks visually appealing:
![image](https://user-images.githubusercontent.com/7101064/73529604-41607b00-4417-11ea-8d51-e554f047f761.png)

Reply:
I agree that it has some appeal, but we already have some inputs documented in other way. `Returns` may also work in the description.
I wouldn't add a second way of documenting inputs unless it can be automated and all of the docs can be aligned with it.